app/src/main/java/com/github/h3lp3rs/h3lp/SettingsActivity.kt

```diff
@@ -0,0 +1,109 @@
+package com.github.h3lp3rs.h3lp
+
+import android.content.Intent
+import android.content.SharedPreferences
+import android.os.Build
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.View
+import android.widget.CheckBox
+import android.widget.TextView
+import androidx.annotation.RequiresApi
+import androidx.appcompat.app.AlertDialog
+import com.github.h3lp3rs.h3lp.signin.GoogleSignInAdapter.getCreationDate
+import com.github.h3lp3rs.h3lp.signin.GoogleSignInAdapter.signOut
+import com.github.h3lp3rs.h3lp.signin.SignInActivity
+import com.github.h3lp3rs.h3lp.signin.SignInActivity.Companion.getUid
+import com.github.h3lp3rs.h3lp.storage.Storages
+import com.github.h3lp3rs.h3lp.storage.Storages.*
+import com.github.h3lp3rs.h3lp.storage.Storages.Companion.storageOf
+import org.apache.commons.collections4.iterators.SkippingIterator
+
+class SettingsActivity : AppCompatActivity() {
+
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_settings)
+        loadSyncPref()
+        findViewById<TextView>(R.id.unique_id_text).text= getUid()
+        findViewById<TextView>(R.id.user_since_text).text= getCreationDate()
+
+    }
+
+    /**
+     * When the user leaves the activity
+     */
+    @RequiresApi(Build.VERSION_CODES.O)
+    override fun onStop() {
```

Comment:
I think this should be an onPause, not onStop so that as soon as the user quits this activity, their changes are saved